core/trino-main/src/test/java/io/trino/server/remotetask/TestHttpRemoteTask.java

```diff
@@ -412,7 +412,8 @@ public void testAdaptiveRemoteTaskRequestSize()
         remoteTask.addSplits(splits);
 
         poll(() -> testingTaskResource.getTaskSplitAssignment(TABLE_SCAN_NODE_ID) != null);
-        poll(() -> testingTaskResource.getTaskSplitAssignment(TABLE_SCAN_NODE_ID).getSplits().size() < 5000); // to check whether the splits are divided or not
+
+        poll(() -> testingTaskResource.getTaskSplitAssignment(TABLE_SCAN_NODE_ID).getSplits().size() < 10000); // to check whether the splits are divided or not
```

Comment:
This still seems potentially flakey to me since the polling interval is coarse grained. Maybe the better assertion would be to count the number of update requests is greater than one and that all splits eventually get delivered?

Reply:
makes sense.